Description: Remove build type check in cmake
Author: Anton Gladky <gladk@debian.org>
Last-Update: 2015-06-16
Index: eigen3/CMakeLists.txt
===================================================================
--- eigen3.orig/CMakeLists.txt
+++ eigen3/CMakeLists.txt
@@ -20,12 +20,6 @@ if (NOT CMAKE_BUILD_TYPE)
 endif()
 
 string(TOLOWER "${CMAKE_BUILD_TYPE}" cmake_build_type_tolower)
-if(    NOT cmake_build_type_tolower STREQUAL "debug"
-   AND NOT cmake_build_type_tolower STREQUAL "release"
-   AND NOT cmake_build_type_tolower STREQUAL "relwithdebinfo")
-  message(FATAL_ERROR "Unknown build type \"${CMAKE_BUILD_TYPE}\". Allowed values are Debug, Release, RelWithDebInfo (case-insensitive).")
-endif()
-
 
 #############################################################################
 # retrieve version infomation                                               #
